### DISCUSSION TITLE
RFC: Increase `Nokogiri::Gumbo::DEFAULT_MAX_ERRORS` to a positive number

### DISCUSSION THREAD

**Discussion by flavorjones** [sparklemotion/nokogiri#3248] [RFCs]
I'm not totally sure why this was initially set to 0 but I think it would be useful to:

- set it to a finite positive number, so some errors are recorded from the parser
- if the parser hits the max, then the last error should be something like `There are more errors which were not recorded. Please set :max_errors to a larger number.`

I would consider eventually setting the default to `-1` to capture all errors, but we should probably think about whether this is something that could be abused by an attacker via untrusted content.

**Comment by stevecheckoway** [comment]
It's a holdover from Nokogumbo. See https://github.com/rubys/nokogumbo/issues/78 for some discussion and https://github.com/rubys/nokogumbo/pull/65 for the initial rationale.

**Comment by flavorjones** [comment]
Fascinating threads, thank you. I guess I'll think about this a bit more.

**Comment by flavorjones** [comment]
I've decided to close this without taking any action. See the linked threads above for context and rationale.
